interpreter/cling/lib/Interpreter/DynamicLibraryManagerSymbol.cpp

```diff
@@ -433,29 +507,99 @@ namespace cling {
         if (!WasInserted)
           continue;
 
-        std::error_code EC;
-        for (llvm::sys::fs::directory_iterator DirIt(DirPath, EC), DirEnd;
-             DirIt != DirEnd && !EC; DirIt.increment(EC)) {
-
-          // FIXME: Use a StringRef here!
-          std::string FileName = getRealPath(DirIt->path());
-          assert(!llvm::sys::fs::is_symlink_file(FileName));
-
+        std::function<const LibraryPath*(std::string, unsigned)> HandleLib =
```

Comment:
```suggestion
        auto HandleLib =
```

Reply:
`auto` only works for non-recursive lambda functions -- we need to fix the return type.

Reply:
Can you use trailing return instead, in the definition of the labmda? `auto` is just nicer (and FWIW actually more performant that having to wrap this in a `std::function`)

Reply:
Nope: https://godbolt.org/z/3hxGs6K9v

Reply:
Oh, *recursive* - missed that (despite you calling it out), thanks!